src/sync.ts

```diff
@@ -1383,27 +1394,16 @@ export class SyncApi {
 
             room.recalculate();
             if (joinObj.isBrandNewRoom) {
-                client.store.storeRoom(room);
                 client.emit(ClientEvent.Room, room);
             }
 
             this.processEventsForNotifs(room, events);
 
-            const processRoomEvent = async (e): Promise<void> => {
-                client.emit(ClientEvent.Event, e);
-                if (e.isState() && e.getType() == "m.room.encryption" && this.opts.crypto) {
-                    await this.opts.crypto.onCryptoEvent(e);
-                }
-            };
-
-            await utils.promiseMapSeries(stateEvents, processRoomEvent);
-            await utils.promiseMapSeries(events, processRoomEvent);
-            ephemeralEvents.forEach(function(e) {
-                client.emit(ClientEvent.Event, e);
-            });
-            accountDataEvents.forEach(function(e) {
-                client.emit(ClientEvent.Event, e);
-            });
+            const emitEvent = (e: MatrixEvent): boolean => client.emit(ClientEvent.Event, e);
```

Comment:
This should make ESLint happy:

```suggestion
            const emitEvent = (e: MatrixEvent): void => client.emit(ClientEvent.Event, e);
```

Reply:
it seemed unhappy with `void`, but happy with `boolean` :man_shrugging: